functions/Get-DbaDbExtentDiff.ps1

```diff
@@ -0,0 +1,273 @@
+
+                WHILE (@extentID < @fileSizePages)
+                BEGIN
+                SELECT @pageID = @extentID + 6;
+
+                SELECT @dbccPageString = 'DBCC PAGE (['
+                + @dbName + '], '
+                + CAST (@fileID AS VARCHAR) + ', '
+                + CAST (@pageID AS VARCHAR) + ', 3) WITH TABLERESULTS, NO_INFOMSGS';
+
+                TRUNCATE TABLE [tempdb].[dbo].[SQLskillsDBCCPage];
+                INSERT INTO [tempdb].[dbo].[SQLskillsDBCCPage] EXEC (@dbccPageString);
+
+                SELECT @total = SUM ([tempdb].[dbo].[SQLskillsConvertToExtents] ([Field]))
+                FROM [tempdb].[dbo].[SQLskillsDBCCPage]
+                WHERE [VALUE] = '    CHANGED'
+                AND [ParentObject] LIKE 'DIFF_MAP%';
+
+                SET @DIFFtotal = @DIFFtotal + @total;
+
+                SET @extentID = @extentID + 511232;
+                END
+
+                FETCH NEXT FROM [files] INTO @fileID, @fileSizePages;
+                END;
+
+                DROP TABLE [tempdb].[dbo].[SQLskillsDBCCPage];
+                CLOSE [files];
+                DEALLOCATE [files];
+
+                SELECT
+                @sizeTotal AS [Total Extents],
+                @DIFFtotal AS [Changed Extents],
+                ROUND (
+                (CONVERT (FLOAT, @DIFFtotal) /
+                CONVERT (FLOAT, @sizeTotal)) * 100, 2) AS [Percentage Changed];
+                END;
+                GO
+            "
+
+            # Extents are a collection of eight physically contiguous pages and are used to efficiently manage the pages.
+            # All pages are stored in extents.
+            $SQLskillsDIFForFULL2017 = "
+                SELECT
+                total_page_count / 8 as [Total Extents],
+                modified_extent_page_count / 8 as [Changed Extents],
+                (100 * modified_extent_page_count)/total_page_count as [Percentage Changed]
+                FROM sys.dm_db_file_space_usage
+                GO
+            "
+
+            if ($server.VersionMajor -ge 14 ) {
+                $queryToRun = $SQLskillsDIFForFULL2017
+            } else {
+                $queryToRun = $SQLskillsDIFForFULL
+            }
+
+            $parsedQuery = @()
+            foreach ($line in $queryToRun) {
+                $line = $line -replace '(^\s+|\s+$)', ''
+                $line = $line -replace "(?m)^\s+"
+                $line = $line -replace "`t", ''
+                $parsedQuery += $line + "`r`n"
+            }
+
+               if (-not $server.Databases[$Database]) {
+                Stop-Function -Message "Database $Database does not exist on $server"
+                return
+            } else {
+                $db = $server.Databases[$Database]
+            }
+
+            # For SQL Server version lower than 2017 we need to create some objects
+            if ($server.VersionMajor -lt 14 ) {
+                try {
+                    foreach ($query in ($parsedQuery -Split "\nGO\b")) {
```

Comment:
invoke-dbasqlcmd parses GO automatically now. However, if you don't need particularly-formatted resultsets and you're not passing parameters to be interpolated, $server.Query() is ok to use.